src/RECONR/ReferenceGrid/src/call.hpp

```diff
@@ -1,65 +1,172 @@
+std::array<double, 3>
+operator()( const resolved::BreitWigner::Lvalue::Resonance& resonance ) const {
+  const double energy = resonance.ER();
+  const double halfWidth = 0.5 * resonance.GT();
+  return {{ energy - halfWidth, energy, energy + halfWidth }};
 }
 
-std::array<double, 3> operator()( 
-  const RP::resolved::BreitWigner::Lvalue::Resonance& resonance ) const {
+std::array<double, 3>
+operator()( const resolved::ReichMoore::Lvalue::Resonance& resonance ) const {
+  const double energy = resonance.ER();
+  const double halfWidth = 0.5 * ( resonance.GN()
+                                   + resonance.GG()
+                                   + resonance.GFA()
+                                   + resonance.GFB() );
+  return {{ energy - halfWidth, energy, energy + halfWidth }};
+}
 
-  const double E = resonance.ER();
-  const double hw = resonance.GT()*0.5;
-  return {{E-hw, E, E+hw}};
+template< typename ResolvedLvalue,
+          if_< hasResonances< ResolvedLvalue > > = true >
+auto
+operator()( const ResolvedLvalue& lValue ) const {
+  return lValue.resonances() | ranges::view::transform( *this );
 }
 
-auto operator()( 
-  const RP::resolved::ReichMoore::Lvalue::Resonance& resonance ) const {
+template< typename Range, if_< hasLvalues< Range > > = true >
+auto operator()( const Range& range ) const {
+  std::vector< double > energies;
+  energies.reserve( 12 * range.lValues().size() );
+
+  const auto lowerLimit = range.EL();
+  const auto upperLimit = nudgeDown( range.EH() );
+
+  energies.push_back( lowerLimit );
+
+  {
+    auto resonances = range.lValues() | ranges::view::for_each( *this );
+    RANGES_FOR( auto resonance, resonances ){
+      ranges::action::push_back( energies,
+                                 resonance
+                                 | ranges::view::filter
+                                   ( [=]( const auto energy )
+                                     { return lowerLimit < energy
+                                              and energy < upperLimit; } ) );
+    }
+  }
 
-  const double E = resonance.ER();
-  const double hw = 0.5*(resonance.GN() + 
-                         resonance.GG() +
-                         resonance.GFA() +
-                         resonance.GFB() );
-  buffer = {{E-hw, E, E+hw}};
-  return ranges::make_iterator_range( buffer.begin(), buffer.end() );
+  energies.push_back( upperLimit );
+
+  if ( not ranges::is_sorted( energies ) ){
+    orlp::pdqsort( energies.begin(), energies.end() );
+  }
+
+  energies |= ranges::action::unique;
+  return energies;
 }
 
-template< typename resolvedL >
-auto operator()( const resolvedL& lValue,
-                 std::conditional_t< true, void*,  decltype( lValue.resonances() ) > = nullptr) const {
+auto operator()( const unresolved::CaseA& caseA ) const {
+  std::vector< double > energies;
+  const auto lowerLimit = caseA.EL();
+  const auto upperLimit = nudgeDown( caseA.EH() );
+  energies.reserve( std::ceil( 5.645828264742274 *
```

Comment:
What is the 5.64... number? Without context, it looks arbitrary.

Reply:
```
log_1.193776641714437( EH / EL ) 
= log( EH / EL ) / log( 1.193776641714437 ) 
= log( EH / EL ) / log( 10^(1/13) )
= 13 * log( EH / EL ) / log( 10 )
= 5.645828264742274 * log( EH / EL )
```

I'll add a comment